pkg/agent/openflow/pipeline.go

```diff
@@ -1119,254 +1252,337 @@ func (c *client) traceflowL2ForwardOutputFlows(dataplaneTag uint8, liveTraffic,
-			Action().GotoTable(L3DecTTLTable.GetID()).
-			Cookie(c.cookieAllocator.Request(category).Raw()).
-			Done()
-		flows = append(flows, flow)
 	}
 	return flows
 }
 
-// l3FwdFlowRouteToPod generates the flows to route the traffic to a Pod based on
-// the destination IP. It rewrites the destination MAC of the packets to the Pod
-// interface MAC. The flow is used in the networkPolicyOnly mode for the traffic
-// from the gateway to a local Pod.
-func (c *client) l3FwdFlowRouteToPod(podInterfaceIPs []net.IP, podInterfaceMAC net.HardwareAddr, category cookie.Category) []binding.Flow {
+// Feature: NetworkPolicy
+// Stage: RoutingStage
+// Tables: L3ForwardingTable
+// Requirements: networkPolicyOnly mode
+// Refactored from:
+//   - func (c *client) l3FwdFlowRouteToPod(podInterfaceIPs []net.IP, podInterfaceMAC net.HardwareAddr, category cookie.Category) []binding.Flow
+// l3FwdFlowRouteToPod generates the flows to route the packets to a Pod based on the destination IP. It rewrites the
+// destination MAC of the packets to the Pod interface MAC. The flow is used in the networkPolicyOnly mode for the packets
+// from Antrea gateway to a local Pod.
+func (c *featureNetworkPolicy) l3FwdFlowRouteToPod(category cookie.Category,
+	podInterfaceIPs []net.IP,
+	podInterfaceMAC net.HardwareAddr) []binding.Flow {
 	var flows []binding.Flow
 	for _, ip := range podInterfaceIPs {
 		ipProtocol := getIPProtocol(ip)
-		flows = append(flows, L3ForwardingTable.BuildFlow(priorityNormal).MatchProtocol(ipProtocol).
+		flows = append(flows, L3ForwardingTable.ofTable.BuildFlow(priorityNormal).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			MatchProtocol(ipProtocol).
 			MatchDstIP(ip).
 			Action().SetDstMAC(podInterfaceMAC).
-			Action().GotoTable(L3DecTTLTable.GetID()).
-			Cookie(c.cookieAllocator.Request(category).Raw()).
+			Action().LoadRegMark(ToLocalRegMark).
+			Action().NextTable().
 			Done())
 	}
 	return flows
 }
 
-// l3FwdFlowRouteToGW generates the flows to route the traffic to the gateway
-// interface. It rewrites the destination MAC of the packets to the gateway
-// interface MAC. The flow is used in the networkPolicyOnly mode for the traffic
-// from a local Pod to remote Pods, Nodes, or external network.
-func (c *client) l3FwdFlowRouteToGW(gwMAC net.HardwareAddr, category cookie.Category) []binding.Flow {
+// Feature: NetworkPolicy
+// Stage: RoutingStage
+// Tables: L3ForwardingTable
+// Requirements: networkPolicyOnly mode
+// Refactored from:
+//   - func (c *client) l3FwdFlowRouteToGW(gwMAC net.HardwareAddr, category cookie.Category) []binding.Flow
+// l3FwdFlowRouteToGW generates the flows to route packets to Antrea gateway. It rewrites the destination MAC of the packets
+// to Antrea gateway interface MAC. The flow is used in the networkPolicyOnly mode for the packets from a local Pod to remote
+// Pods, Nodes, or external network.
+func (c *featureNetworkPolicy) l3FwdFlowRouteToGW(category cookie.Category) []binding.Flow {
```

Comment:
I think this flow is not in the scope of PodConnectivity, but not about NetworkPolicy, although it works in NetworkPolicyOnly mode.

So is `arpResponderStaticFlow`

Reply:
Updated